### ISSUE TITLE
Swiper match face dot shown

### ISSUE THREAD

**Issue by Ergus** [abo-abo/swiper#1928]
Hi:
This is a small issue, but any way I considered important to report it.

After the fixes of #1921 and #1923, now the match part of the candidate is not highlighted when the match was outside the screen (in the terminal). The line is highlighted properly, but the match part not. The candidates that are already on the screen have the match part highlighted right, and when they are selected the match remain highlighted properly and the one that was not, becomes right. This always happen when the candidate is outside the screen in the terminal.

**Comment by abo-abo** [comment]
I don't understand. Please add a small example to reproduce.

**Comment by basil-conto** [comment]
1. `make emacs='emacs -nw' plain`
2. <kbd>C-x</kbd><kbd>C-f</kbd>`ivy.el`<kbd>RET</kbd>
3. <kbd>C-s</kbd>`ivy`
4. Repeat <kbd>C-n</kbd> until a candidate that was not originally visible in the window is selected.

   ![4-redisplay](https://user-images.githubusercontent.com/9121222/52631304-61b9c180-2eb6-11e9-8a60-9685ae63a44a.png)

   Note how `ivy-read`, `ivy-next-line`, and `ivy-previous-line` are fontified in the minibuffer but not in the original buffer.

**Comment by abo-abo** [comment]
Thanks, all. Please test.

**Comment by basil-conto** [comment]
Thanks, that's fixed the fontification issue for me. I was never able to reproduce the flashing that @Ergus reported (even with `recenter-redisplay` set to its default `tty`), though, so their testing is more important than mine.

**Comment by Ergus** [comment]
Yes, this also fixed it for me. Very thanks!! 
On the other hand @basil-conto the problem affects me because I use emacs in the terminal (xterm). I you don't have the problem in a terminal, please, tell me what emmulator do you use to try it.

**Comment by basil-conto** [comment]
> the problem affects me because I use emacs in the terminal (xterm)

I know. I too was using Emacs under a text terminal for the purposes of this issue, both Emacs 26 and latest `master`.

> tell me what emmulator do you use to try it

`gnome-terminal`

**Comment by Ergus** [comment]
I just tried gnome-terminal for this and the flash when recentering (C-l) is still there (I am in emacs 26.1 from arch linux). Except of course if I set recenter-redisplay to nil. 
This issue was in fact reported and fixed in [bug#31325](http://lists.gnu.org/archive/html/bug-gnu-emacs/2018-06/msg01108.html) so maybe the fixed will be not needed after emacs 27 will be released.
In my huge ignorance I would like to understand why:
1) Redisplay flickers so notably for me.
2) Why recenter works perfectly without redisplay.
3) What specific  situation makes that recenter comes with redisplay by default (or why).
But I just started reading the emacs display engine code...
